Source/Core/DolphinQt/CheatsManager.cpp

```diff
@@ -172,14 +148,9 @@ void CheatsManager::OnNewSessionCreated(const Cheats::CheatSearchSessionBase& se
 {
   auto* w = new CheatSearchWidget(m_system, session.Clone());
   const int tab_index = m_tab_widget->addTab(w, tr("Cheat Search"));
-  w->connect(w, &CheatSearchWidget::ActionReplayCodeGenerated, this,
-             [this](const ActionReplay::ARCode& ar_code) {
-               if (m_ar_code)
-                 m_ar_code->AddCode(ar_code);
-             });
-  w->connect(w, &CheatSearchWidget::ShowMemory, [this](u32 address) { emit ShowMemory(address); });
-  w->connect(w, &CheatSearchWidget::RequestWatch,
-             [this](QString name, u32 address) { emit RequestWatch(name, address); });
+  connect(w, &CheatSearchWidget::ActionReplayCodeGenerated, m_ar_code, &ARCodeWidget::AddCode);
+  connect(w, &CheatSearchWidget::ShowMemory, this, &CheatsManager::ShowMemory);
+  connect(w, &CheatSearchWidget::RequestWatch, this, &CheatsManager::RequestWatch);
```

Comment:
I'm not sure if I agree with directly calling signals as if they were slots. If I understand, the `emit` keyword is syntactic sugar and nothing more, but it's still helpful self-documentation.

Reply:
The [Qt docs](https://doc.qt.io/qt-6/qobject.html#connect) say "A signal can also be connected to another signal" and give this (abbreviated) example:

```
myButton = new QPushButton(this);
connect(myButton, SIGNAL(clicked()), this, SIGNAL(buttonClicked()));
```

If we want to send a signal manually (i.e. not in a `connect` call) then I agree with using the `emit` keyword for clarity even though it's not necessary, but for `connect` I think wrapping it in a lambda just to use `emit` would make more noise than light.

Reply:
Well I can't argue with the official docs. I learn a new thing about Qt every day.